src/directives/index.ts

```diff
@@ -72,14 +73,15 @@ const parseArguments = (directives: Directives): RuleCore => (state) => {
       const directive = getDirective(directives, token.attrGet('kind'));
       if (!match || !directive) throw new Error('Shoud not be able to get into here without matching?');
       const info = match[2].trim();
-      const { attrs, content: modified } = directive.getArguments?.(info) ?? {};
-      Object.entries(attrs ?? {}).map(([k, v]) => token.attrSet(k, v));
+      const { args, content: modified } = directive.getArguments?.(info) ?? {};
+      token.meta = { ...token.meta, args };
```

Comment:
Args/opts data is on the `token.meta` and this is shared in both the opening and closing tags.